internal/resource/nvml-device.go

```diff
@@ -99,3 +101,35 @@ func (d nvmlDevice) GetPCIClass() (uint32, error) {
 	}
 	return nvDevice.Class, nil
 }
+
+func (d nvmlDevice) GetClusterUUID() (string, error) {
+	gfInfo, ret := d.GetGpuFabricInfo()
+	if ret != nvml.SUCCESS {
+		return "", ret
+	}
+
+	// Convert the array to a byte slice
+	byteSlice := gfInfo.ClusterUuid[:]
```

Comment:
```suggestion
	byteSlice := gfInfo.ClusterUUID[:]
```

Reply:
I don't control this one, it is coming directly from nvml https://github.com/NVIDIA/go-nvml/blob/main/pkg/nvml/types_gen.go#L681

Reply:
Ack